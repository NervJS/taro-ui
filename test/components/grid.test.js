import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'
import AtGrid from '../../.temp/components/grid/index'

const IMAGE_DATA = [
  {
    image:
      'https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png',
    value: '领取中心'
  },
  {
    image:
      'https://img20.360buyimg.com/jdphoto/s72x72_jfs/t15151/308/1012305375/2300/536ee6ef/5a411466N040a074b.png',
    value: '找折扣'
  },
  {
    image:
      'https://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png',
    value: '领会员'
  },
  {
    image:
      'https://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png',
    value: '新品首发'
  },
  {
    value: '支持Icon',
    iconInfo: {
      size: 30,
      color: 'red',
      value: 'bookmark'
    }
  }
]

const ICON_DATA = [
  {
    icon: 'bell',
    value: '领取中心'
  },
  {
    icon: 'bookmark',
    value: '找折扣'
  },
  {
    icon: 'calendar',
    value: '领会员'
  },
  {
    icon: 'bell',
    value: 'subtract'
  }
]

describe('Grid Snap', () => {
  it('render image Grid', () => {
    const component = renderToString(<AtGrid data={IMAGE_DATA} />)
    expect(component).toMatchSnapshot()
  })

  it('render no border of image Grid', () => {
    const component = renderToString(<AtGrid data={IMAGE_DATA} />)
    expect(component).toMatchSnapshot()
  })

  it('render rect image Grid', () => {
    const component = renderToString(<AtGrid mode='rect' data={IMAGE_DATA} />)
    expect(component).toMatchSnapshot()
  })

  it('render icon Grid', () => {
    const component = renderToString(<AtGrid data={ICON_DATA} />)
    expect(component).toMatchSnapshot()
  })

  it('render rect icon Grid', () => {
    const component = renderToString(<AtGrid mode='rect' data={ICON_DATA} />)
    expect(component).toMatchSnapshot()
  })

  it('render rect icon Grid -- props columnNum', () => {
    const component = renderToString(
      <AtGrid mode='rect' columnNum={4} data={ICON_DATA} />
    )
    expect(component).toMatchSnapshot()
  })
})

describe('Grid Behavior ', () => {
  it('Grid onClick', () => {
    const onClick = jest.fn()

    const component = renderIntoDocument(
      <AtGrid mode='rect' columnNum={4} data={ICON_DATA} onClick={onClick} />
    )
    const componentDom = findDOMNode(component, 'at-grid')
    const itemDom = componentDom.querySelector('.at-grid-item')

    Simulate.click(itemDom)
    expect(onClick).toBeCalled()
  })
})
