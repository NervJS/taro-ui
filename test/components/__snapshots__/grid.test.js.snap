// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Snap render icon Grid 1`] = `"<div class=\\"at-grid\\"><div class=\\"at-grid__flex\\"><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefinedundefined</div><span class=\\"taro-text content-inner__text\\">领取中心</span></div></div></div><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefinedundefined</div><span class=\\"taro-text content-inner__text\\">找折扣</span></div></div></div><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square at-grid-item--last\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefinedundefined</div><span class=\\"taro-text content-inner__text\\">领会员</span></div></div></div></div><div class=\\"at-grid__flex\\"><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefinedundefined</div><span class=\\"taro-text content-inner__text\\">subtract</span></div></div></div></div></div>"`;

exports[`Grid Snap render image Grid 1`] = `"<div class=\\"at-grid\\"><div class=\\"at-grid__flex\\"><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\"><div class=\\"taro-img content-inner__img\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png\\"/></div>undefined</div><span class=\\"taro-text content-inner__text\\">领取中心</span></div></div></div><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\"><div class=\\"taro-img content-inner__img\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"https://img20.360buyimg.com/jdphoto/s72x72_jfs/t15151/308/1012305375/2300/536ee6ef/5a411466N040a074b.png\\"/></div>undefined</div><span class=\\"taro-text content-inner__text\\">找折扣</span></div></div></div><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square at-grid-item--last\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\"><div class=\\"taro-img content-inner__img\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"https://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png\\"/></div>undefined</div><span class=\\"taro-text content-inner__text\\">领会员</span></div></div></div></div><div class=\\"at-grid__flex\\"><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\"><div class=\\"taro-img content-inner__img\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"https://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png\\"/></div>undefined</div><span class=\\"taro-text content-inner__text\\">新品首发</span></div></div></div><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefined<span style=\\"color:red;font-size:30px;\\" class=\\"taro-text at-icon at-icon-bookmark\\"></span></div><span class=\\"taro-text content-inner__text\\">支持Icon</span></div></div></div></div></div>"`;

exports[`Grid Snap render no border of image Grid 1`] = `"<div class=\\"at-grid\\"><div class=\\"at-grid__flex\\"><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\"><div class=\\"taro-img content-inner__img\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png\\"/></div>undefined</div><span class=\\"taro-text content-inner__text\\">领取中心</span></div></div></div><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\"><div class=\\"taro-img content-inner__img\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"https://img20.360buyimg.com/jdphoto/s72x72_jfs/t15151/308/1012305375/2300/536ee6ef/5a411466N040a074b.png\\"/></div>undefined</div><span class=\\"taro-text content-inner__text\\">找折扣</span></div></div></div><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square at-grid-item--last\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\"><div class=\\"taro-img content-inner__img\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"https://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png\\"/></div>undefined</div><span class=\\"taro-text content-inner__text\\">领会员</span></div></div></div></div><div class=\\"at-grid__flex\\"><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\"><div class=\\"taro-img content-inner__img\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"https://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png\\"/></div>undefined</div><span class=\\"taro-text content-inner__text\\">新品首发</span></div></div></div><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--square\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefined<span style=\\"color:red;font-size:30px;\\" class=\\"taro-text at-icon at-icon-bookmark\\"></span></div><span class=\\"taro-text content-inner__text\\">支持Icon</span></div></div></div></div></div>"`;

exports[`Grid Snap render rect icon Grid -- props columnNum 1`] = `"<div class=\\"at-grid\\"><div class=\\"at-grid__flex\\"><div style=\\"flex:0 0 25%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--rect\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefinedundefined</div><span class=\\"taro-text content-inner__text\\">领取中心</span></div></div></div><div style=\\"flex:0 0 25%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--rect\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefinedundefined</div><span class=\\"taro-text content-inner__text\\">找折扣</span></div></div></div><div style=\\"flex:0 0 25%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--rect\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefinedundefined</div><span class=\\"taro-text content-inner__text\\">领会员</span></div></div></div><div style=\\"flex:0 0 25%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--rect at-grid-item--last\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefinedundefined</div><span class=\\"taro-text content-inner__text\\">subtract</span></div></div></div></div></div>"`;

exports[`Grid Snap render rect icon Grid 1`] = `"<div class=\\"at-grid\\"><div class=\\"at-grid__flex\\"><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--rect\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefinedundefined</div><span class=\\"taro-text content-inner__text\\">领取中心</span></div></div></div><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--rect\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefinedundefined</div><span class=\\"taro-text content-inner__text\\">找折扣</span></div></div></div><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--rect at-grid-item--last\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefinedundefined</div><span class=\\"taro-text content-inner__text\\">领会员</span></div></div></div></div><div class=\\"at-grid__flex\\"><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--rect\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefinedundefined</div><span class=\\"taro-text content-inner__text\\">subtract</span></div></div></div></div></div>"`;

exports[`Grid Snap render rect image Grid 1`] = `"<div class=\\"at-grid\\"><div class=\\"at-grid__flex\\"><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--rect\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\"><div class=\\"taro-img content-inner__img\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"https://img12.360buyimg.com/jdphoto/s72x72_jfs/t6160/14/2008729947/2754/7d512a86/595c3aeeNa89ddf71.png\\"/></div>undefined</div><span class=\\"taro-text content-inner__text\\">领取中心</span></div></div></div><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--rect\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\"><div class=\\"taro-img content-inner__img\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"https://img20.360buyimg.com/jdphoto/s72x72_jfs/t15151/308/1012305375/2300/536ee6ef/5a411466N040a074b.png\\"/></div>undefined</div><span class=\\"taro-text content-inner__text\\">找折扣</span></div></div></div><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--rect at-grid-item--last\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\"><div class=\\"taro-img content-inner__img\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"https://img10.360buyimg.com/jdphoto/s72x72_jfs/t5872/209/5240187906/2872/8fa98cd/595c3b2aN4155b931.png\\"/></div>undefined</div><span class=\\"taro-text content-inner__text\\">领会员</span></div></div></div></div><div class=\\"at-grid__flex\\"><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--rect\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\"><div class=\\"taro-img content-inner__img\\"><img class=\\"taro-img__mode-scaletofill\\" src=\\"https://img12.360buyimg.com/jdphoto/s72x72_jfs/t10660/330/203667368/1672/801735d7/59c85643N31e68303.png\\"/></div>undefined</div><span class=\\"taro-text content-inner__text\\">新品首发</span></div></div></div><div style=\\"flex:0 0 33.333333333333336%;\\" class=\\"at-grid__flex-item at-grid-item at-grid-item--rect\\"><div class=\\"at-grid-item__content\\"><div class=\\"at-grid-item__content-inner\\"><div class=\\"content-inner__icon\\">undefined<span style=\\"color:red;font-size:30px;\\" class=\\"taro-text at-icon at-icon-bookmark\\"></span></div><span class=\\"taro-text content-inner__text\\">支持Icon</span></div></div></div></div></div>"`;
